main.c

```diff
@@ -296,11 +359,20 @@ int main(int argc, char *argv[]) {
 		}
 	}
 
+	char *output_filename;
+	char tmp[64];
 	if (optind >= argc) {
-		fprintf(stderr, "%s", usage);
-		return EXIT_FAILURE;
+		if (default_filename(tmp, sizeof(tmp), output_filetype) != true) {
+			fprintf(stderr, "failed to generate default filename\n");
+			return EXIT_FAILURE;
+		}
+		output_filename = tmp;
+	} else {
+		output_filename = argv[optind];
 	}
-	char *output_filename = argv[optind];
+
+	char output_filepath[PATH_MAX];
+	filepath(output_filepath, output_filename);
```

Comment:
One last comment: in case a filename is specified on the CLI, we don't want to prepend our default directory.